components/eventing-controller/controllers/subscription/nats/reconciler.go

```diff
@@ -225,21 +225,21 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	// Check for valid sink
 	if err := r.sinkValidator(ctx, r, subscription); err != nil {
 		log.Errorw("sink URL validation failed", "error", err)
-		if err := r.syncSubscriptionStatus(ctx, subscription, false, false, err.Error()); err != nil {
-			return checkIsConflict(err)
+		if syncErr := r.syncSubscriptionStatus(ctx, subscription, false, false, err.Error()); err != nil {
+			return ctrl.Result{}, syncErr
```

Comment:
The method `checkIsConflict` was implemented in a previous [PR](https://github.com/kyma-project/kyma/pull/13070/files):

```go
func checkIsConflict(err error) (ctrl.Result, error) {
	if k8serrors.IsConflict(err) {
		// Requeue the Request to try to reconcile it again
		return ctrl.Result{Requeue: true}, nil
	}
	return ctrl.Result{}, err
}
```
As you described in the issue already, there is no fundamental difference between 
`return ctrl.Result{}, err` and `return ctrl.Result{Requeue: true}, nil`. 

The latter will miss these two lines:

```go
ctrlmetrics.ReconcileErrors.WithLabelValues(c.Name).Inc()
log.Error(err, "Reconciler error")
```

So the basic question is if we want to treat `k8serrors.IsConflict` as an expected error. If yes, then we don't need to log it out and don't need to increase the metrics for errored reconciliations.

My personal opinion is that it is nice to not bloat the logs of the reconciler with the conflict error. That is just noise too my eyes then debugging something.

What is your take ?

Reply:
cc: @radufa since you implemented the change.

Reply:
I agree that as it is, here `checkIsConflict` doesn't really make much of a difference aside from causing the controller runtime to also log the error. In some cases I've removed it (when subscription is not ready). In other cases, I'm not sure if it can be removed.

If you're only concern is bloated logs (which I also dislike), we need to do bigger clean ups. Currently, whenever we encounter an error in the reconciler, we log it and we return an error. Returning an error would also get logged by the controller runtime. So I think, we shouldn't log just the error, if we have no other extra information (we could return an enriched/wrapped error and leave the logging to the controller runtime).

Reply:
>My personal opinion is that it is nice to not bloat the logs of the reconciler with the conflict error. That is just noise too my eyes then debugging something.

Agree, for the "helping the customer" perspective, is good to have in the log file only "real errors". The customer is reading the log file and send it somehow to us for the future analysis. We have tried to clean-up the log file (in main), so that under "info" only errors should be logged after the controller is stabilized.

Reply:
@pxsalehi I agree that logging an error and returning it to controller-runtime to log it as well makes no sense. Your proposed solution makes sense to me :) Would you like to create an issue for it ? :)

Thanks for explaining the pattern you used for removing checkIsConflict. I did not see it before.

I think in both cases we have a log line which states the actual error. So independent of if we (1) log it ourselves or (2) let controller-runtime log it I think we should avoid logging the conflict error in case of a kubernetes conflict.

Reply:
@radufa I think we should add a comment to describe the `advantage` of using `checkIsConflict`.

Reply:
A reminder to clean up the duplicate error log entries: https://github.com/kyma-project/kyma/issues/13219